third_party/proxy/service.go

```diff
@@ -415,6 +419,10 @@ func (sct *ServiceChangeTracker) serviceToServiceMap(service *v1.Service) Servic
 		return nil
 	}
 
+	if !sct.serviceLabelSelector.Matches(labels.Set(service.Labels)) {
```

Comment:
@tnqn

Reply:
I think it should be moved into `ShouldSkipService` to make code more focused

Reply:
Done.